datafusion/execution/src/memory_pool/proxy.rs

```diff
@@ -84,7 +84,7 @@ impl<T> RawTableAllocExt for RawTable<T> {
             Err(x) => {
                 // need to request more memory
 
-                let bump_elements = (self.capacity() * 2).max(16);
+                let bump_elements = self.capacity().max(16);
                 let bump_size = bump_elements * std::mem::size_of::<T>();
                 *accounting = (*accounting).checked_add(bump_size).expect("overflow");
 
```

Comment:
I double checked that the call to `reserve` below ensures *additional* capacity, not *total* capacity which is what the existing code appears to do. 

https://docs.rs/hashbrown/latest/hashbrown/raw/struct.RawTable.html#method.reserve

cc @crepererum  and @Dandandan

Reply:
That was probably my mistake :facepalm: 

Grow factor 200%, but the addition 100%. :+1: